src/System.Private.ServiceModel/src/System/IdentityModel/Claims/X509CertificateClaimSet.cs

```diff
@@ -380,52 +379,108 @@ public IIdentity Identity
         // to figure out what the identifier, delimiter, and separator is by using a well-known extension
         private static class X509SubjectAlternativeNameConstants
         {
-            public const string Oid = "2.5.29.17"; 
-            public static readonly string Identifier;
-            public static readonly char Delimiter;
-            public static readonly string Separator;
+            public const string Oid = "2.5.29.17";
+
+            private static readonly string _identifier;
+            private static readonly char _delimiter;
+            private static readonly string _separator;
+
+            private static bool _successfullyInitialized = false;
+            private static Exception _initializationException;
+
+            public static string Identifier
+            {
+                get
+                {
+                    EnsureInitialized();
+                    return _identifier;
+                }
+            }
+
+            public static char Delimiter
+            {
+                get
+                {
+                    EnsureInitialized();
+                    return _delimiter;
+                }
+            }
+            public static string Separator
+            {
+                get
+                {
+                    EnsureInitialized();
+                    return _separator;
+                }
+            }
+
+            private static void EnsureInitialized()
+            {
+                if (!_successfullyInitialized)
+                {
+                    throw new FormatException(string.Format(
+                        "There was an error detecting the identifier, delimiter, and separator for X509CertificateClaims on this platform.{0}Detected values were: Identifier: '{1}'; Delimiter:'{2}'; Separator:'{3}'",
```

Comment:
Nit, string is very long. Suggest changing to:
```csharp
    "There was an error detecting the identifier, delimiter, and separator for X509CertificateClaims on this platform." +
    Environment.NewLine + "Detected values were: Identifier: '{0}'; Delimiter:'{1}'; Separator:'{2}'",
    _identifier,
    _delimiter,
    _separator
```
For easier readability. Not a required change as this is a style thing.